packages/snaps-controllers/src/snaps/SnapController.ts

```diff
@@ -2777,14 +2784,20 @@ export class SnapController extends BaseController<
       }
     }
 
-    const handler = await this.#getRpcRequestHandler(snapId);
+    const handler = this.#getRpcRequestHandler(snapId);
     if (!handler) {
       throw new Error(
         `Snap RPC message handler not found for snap "${snapId}".`,
       );
     }
 
-    return handler({ origin, handler: handlerType, request });
+    const timeout = this.#getExecutionTimeout(handlerPermissions);
+
+    return handler({ origin, handler: handlerType, request, timeout });
+  }
+
+  #getExecutionTimeout(permission?: PermissionConstraint): number {
```

Comment:
I would suggest adding docstring to this one.

Reply:
Yes! Good catch.